x/gamm/pool-models/internal/cfmm_common/lp.go

```diff
@@ -152,18 +152,20 @@ func BinarySearchSingleAssetJoin(
 			return sdk.Int{}, err
 		}
 
-		return swapAllCoinsToSingleAsset(poolWithUpdatedLiquidity, ctx, exitedCoins, swapToDenom)
+		return SwapAllCoinsToSingleAsset(poolWithUpdatedLiquidity, ctx, exitedCoins, swapToDenom)
 	}
 	// TODO: Come back and revisit err tolerance
```

Comment:
Can we have an issue referenced for the TODO -- this is how tech debt is introduced :p

Reply:
I actually ended up resolving this so I think we can remove this TODO! But agreed we should go through our codebase and create issues for any important TODOs that are lingering